tests/basic/basic_test.go

```diff
@@ -191,9 +192,19 @@ var _ = Describe("{VolumeDriverAppDown}", func() {
 					Expect(appNodes).NotTo(BeEmpty())
 				})
 
+				//get the nodes to be down randomly
+				appNodesLen := len(appNodes)
+				amountAllowedNodesDown := len(node.GetWorkerNodes()) % appNodesLen
```

Comment:
You need to handle below scenario too

- 5 node cluster
- 2 appNodes, node1 and node2
- The volumes is repl=2 and node1 and node2 are both the replicas of the volume

The current code will allow node1 and node1 to do down and hence all nodes for that volume will be down. We don't want that now. We will add a out of quorum test later on.

Also below logic where you are populating nodesToBeDown can be better. You are doing

```
nodesToBeDown = nodesToBeDown[:amountAllowedNodesDown]
```

But amountAllowedNodesDown can be greater than actual appNodes. So you only need to copy upto the min(appNodes,  amountAllowedNodesDown)

Reply:
you're right, it's not taking into account replication factor and if all the nodes with replicas are down